protocols/grpc/agent.proto

```diff
@@ -426,3 +427,10 @@ message StringUser {
 	string gid = 2;
 	repeated string additionalGids = 3;
 }
+
+message CopyFileRequest {
+	string path = 1;
```

Comment:
Please add `offset` and `length` here so that it is possible to send just part of a file. Also please put `content` at last of the struct.

What happens when the file size exceeds the grpc payload limit? If we support offset and length, client can send a file by parts.

Reply:
@bergwolf 
I think length is not needed, since we can get it with `len(content)`, is that correct?

Reply:
@devimc good point!

Reply:
Is this the *absolute* guest root context path? In other words, is this API allowing changes to, say, `/sbin/init`? Or is it designed purely to copy files into the workloads rootfs directory?

If it's possible to overwrite the init binary, we'd better disallow that for now until the agent is able to serialise it's state and re-exec itself.

Reply:
- What about adding a `DoNotOverwrite bool` so the client can change the default behaviour of overwriting an existing file? It could be called `NoClobber bool` I guess, but "clobber" is not a common word even if English is your native tongue ;)

But maybe the default should be to not overwrite? What is the intent of this change?

Reply:
:nerd_face: according with g,xA clobber means *hit (someone) hard. "if he does that I'll clobber him!"*

Reply:
Yeah, probably best not to use that term! No NoClobber I say! :smile:

Reply:
I'm even okay with the overwrite just happening, if it keeps it simple.  You can do similar things today with docker CLI, afaiu, right?  Either way - adding overwrite option seems reasonable to me.